subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java

```diff
@@ -194,6 +197,16 @@ private void configureCookieSpecRegistry(HttpClientBuilder builder) {
         );
     }
 
+    private void configureRequestConfig(HttpClientBuilder builder) {
+        HttpTimeoutSettings timeoutSettings = httpSettings.getTimeoutSettings();
+        RequestConfig config = RequestConfig.custom()
+            .setConnectTimeout(timeoutSettings.getConnectionTimeoutMs())
+            .setSocketTimeout(timeoutSettings.getSocketTimeoutMs())
+            .build();
+        builder.setDefaultRequestConfig(config);
+        builder.setDefaultSocketConfig(SocketConfig.custom().setSoTimeout(timeoutSettings.getSocketTimeoutMs()).build());
```

Comment:
:x: You might want to break this out into a separate method. The method name doesn't align with what it does.

Reply:
Sure, good catch. Done.